api/controllers/linux/install/controller.go

```diff
@@ -265,6 +268,24 @@ func NewInstallController(opts ...InstallControllerOption) (*InstallController,
 			appcontroller.WithAirgapBundle(controller.airgapBundle),
 			appcontroller.WithPrivateCACertConfigMapName(adminconsole.PrivateCASConfigMapName), // Linux installations use the ConfigMap
 		)
+
+		// Detect registry settings and pass to app controller
+		var license *kotsv1beta1.License
+		if controller.license != nil {
+			// Parse license from bytes
```

Comment:
nit: i think there's a helper method to parse the license which can make this a bit cleaner.